main/commands/combineVideos.ts

```diff
@@ -10,22 +10,19 @@ export async function combineVideos(
   outputFilePath: string,
   notifyEvent?: EventEmitter
 ): Promise<void> {
-  winston.info('Generating videoList file');
-  notifyEvent && notifyEvent.emit('Generating videoList file');
-  tmp.setGracefulCleanup();
-  // Create a temporary list file to concatenate videos (see https://trac.ffmpeg.org/wiki/Concatenate)
-  const { path: file, cleanup } = await tmp.file({ postfix: '.txt' });
-  const contents = videoPaths.map((videoPath) => `file '${videoPath}'`).join('\n');
-  writeFileSync(file, contents);
   winston.info('Combining Videos');
   notifyEvent && notifyEvent.emit('Combining Videos');
-  const combineProcess = spawnSync(
-    paths.ffmpeg,
-    ['-f', 'concat', '-loglevel', 'error', '-safe', '0', '-i', file, '-c', 'copy', outputFilePath],
-    {
-      stdio: 'pipe',
-    }
-  );
+  const args: string[] = ['-v', 'error'];
+  const filterComplex: string[] = [];
+  for (let i = 0; i < videoPaths.length; i++) {
+    args.push('-i', videoPaths[i]);
+    filterComplex.push(`[${i}:0]`, `[${i}:1]`);
+  }
+  filterComplex.push(`concat=n=${videoPaths.length}:v=1:a=1`, '[v]', '[a]');
+  args.push('-filter_complex', filterComplex.join(' '), '-map', '[v]', '-map', '[a]', outputFilePath);
+  const combineProcess = spawnSync(paths.ffmpeg, args, {
+    stdio: 'pipe',
+  });
```

Comment:
This is great work @JCFarrow !  I would like to see some comments that explain these lines, at the very least links to SO or FFMPEG documentation that explain what the flags do.  `-filter_complex` is, well, complex and for the uninitiated I think it would be good for your explain what the flags are doing.

Reply:
Good idea. I added a link to the documentation for the concat filter which has the example that I followed.